File: backend/core/project_types.py
"""Central project type constants for VISTA.

The canonical values are persisted/API contract values. User-facing deployment labels
are centralized in ``frontend/src/projectTypes.js``.
"""

PROJECT_TYPES = ("PT1", "PT2", "PT3")
DEFAULT_PROJECT_TYPE = PROJECT_TYPES[0]
PROJECT_TYPE_PATTERN = r"^(PT1|PT2|PT3)$"


def normalize_project_type(project_type: str | None) -> str:
    normalized = str(project_type or DEFAULT_PROJECT_TYPE).strip().upper()
    return normalized if normalized in PROJECT_TYPES else DEFAULT_PROJECT_TYPE

File: frontend/src/projectTypes.js
// Centralized project type display metadata.
// Update these labels when a deployment needs different user-facing names for PT1/PT2/PT3.
export const PROJECT_TYPE_OPTIONS = [
  { value: 'PT1', label: 'PT1 — External Multi-View', shortLabel: 'PT1' },
  { value: 'PT2', label: 'PT2 — 3D Slice Review', shortLabel: 'PT2' },
  { value: 'PT3', label: 'PT3 — Advanced 3D Slice Review', shortLabel: 'PT3' },
];

export const DEFAULT_PROJECT_TYPE = PROJECT_TYPE_OPTIONS[0].value;

export function getProjectTypeLabel(projectType, { short = false } = {}) {
  const option = PROJECT_TYPE_OPTIONS.find((item) => item.value === projectType);
  if (!option) return projectType || DEFAULT_PROJECT_TYPE;
  return short ? option.shortLabel : option.label;
}
